src/utils/comms.ts

```diff
@@ -3,20 +3,20 @@ export default {
      * Posts the provided msg to the specified origin and parent window when invoked within the
      * expected environment (e.g. a frame hosted within PureCloud).
      *
-     * @param {*} msg - The payload to send as a postMessage
-     * @param {*} tgtOrigin - The destination origin (see postMessage)
-     * @param {*} transfer - Transferable objects (see postMessage)
-     * @param {*} myWindow - The current execution window. Default window
-     * @param {*} myParent - The parent window to which the message should be posted. Default parent
-     * @param {object} myConsole - logs errors here rather than throwing errors. Default console
+     * @param msg - The payload to send as a postMessage
+     * @param tgtOrigin - The destination origin (see postMessage)
+     * @param transfer - Transferable objects (see postMessage)
+     * @param myWindow - The current execution window. Default window
+     * @param myParent - The parent window to which the message should be posted. Default parent
+     * @param myConsole - logs errors here rather than throwing errors. Default console
      *
      * @throws Error if the environment is invalid (e.g. not a browser, no postMessage api,
      *  not running within an iframe) and myConsole is not specified
      */
-    postMsgToPc(msg, tgtOrigin, transfer, myWindow = window, myParent = window.parent, myConsole = window.console) {
-        let validRuntime = !!(myWindow && typeof myWindow === 'object' && myParent && typeof myParent === 'object');
-        let validEnv = (validRuntime && myParent !== myWindow);
-        let validApi = !!(myParent && typeof myParent.postMessage === 'function');
+    postMsgToPc(msg: any, tgtOrigin: string, transfer?: Transferable[], myWindow: Window = window, myParent = window.parent, myConsole: Console | null = window.console) {
```

Comment:
Thanks for pulling in the proper postMessage types!  Looks good.

Reply:
Was going to ask why you make the console nullable and then saw the tests.  I am def fine with it staying be nullable, but we could change the tests to use undefined if you felt strongly about not having it nullable.

Also, not exactly sure how TS deals with it with the fact that impls vary amongst browsers.  For instance, I think <IE10 had window.console undefined if devtools weren't open.

Anyway, maybe best not to open this cat basket?

Reply:
Think this could potentially fall under that second pass I described in [this comment](https://github.com/MyPureCloud/client-app-sdk/pull/72#discussion_r473000087) as well. But yeah I'm not sure about IE10 but happy to look into it when we come in to clean this up.